tests_integration/collect_test_scenarios.py

```diff
@@ -0,0 +1,93 @@
+import collections
+import os
+import random
+
+from tests_integration.lib.shared import DEFINITIONS_DIR, ENV_GITHUB_RUN_ID
+
+SKIP_TEST_CASES = {"__init__"}
+
+
+def collect_integration_test_cases() -> dict[str, list[str]]:
+    """
+    For each feedstock, return a list of all test cases that should be run for it.
+    The test cases do not include the feedstock name or the .py extension.
+
+    Example return value:
+    {
+        "feedstock1": ["aarch_migration", "version_update"],
+        "feedstock2": ["version_update"],
+    }
+
+    The return value of this function is deterministic (sorted by feedstock name and test case name).
+    """
+    test_cases = collections.defaultdict(list)
+
+    for test_case in DEFINITIONS_DIR.glob("*/*.py"):
+        test_case_name = test_case.stem
+        if test_case_name in SKIP_TEST_CASES:
+            continue
+        feedstock = test_case.parent.name
+        test_cases[feedstock].append(test_case_name)
+
+    return dict(
+        sorted(
+            (feedstock, sorted(test_cases))
+            for feedstock, test_cases in test_cases.items()
+        )
+    )
+
+
+def get_number_of_test_scenarios(integration_test_cases: dict[str, list[str]]) -> int:
+    return max(len(test_cases) for test_cases in integration_test_cases.values())
```

Comment:
Isn't this supposed to be the sum?

Reply:
No. Per scenario, we can only run one test case per feedstock, but two test cases for different feedstocks can always run in parallel. So the number of test scenarios is the maximum number of test cases of a feedstock.

If a feedstock has less than this number of test cases, it will just repeat its test cases while we iterate the feedstock with the most test cases. See also my answer to your comment [here](https://github.com/regro/cf-scripts/pull/3417#discussion_r1956118310).

Reply:
See the docs here now (new): https://github.com/ytausch/cf-scripts/tree/integration-tests/tests_integration